client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-set-favorite/index.tsx

```diff
@@ -0,0 +1,158 @@
+
+	const handleViewFavorites = () => {
+		dispatch( removeNotice( successNoticeId ) );
+		page.redirect( '/dashboard/favorites?highlight=favorite-tab' );
+	};
+
+	const handleOnChangeFavoriteSuccess = () => {
+		const text = (
+			<span>
+				{ ! isFavorite
+					? translate( '%(siteUrl)s has been added to your favorites.', {
+							args: {
+								siteUrl,
+							},
+					  } )
+					: translate( '%(siteUrl)s has been removed from your favorites.', {
+							args: {
+								siteUrl,
+							},
+					  } ) }
+				{ ! showOnlyFavorites && (
+					<Button
+						className="site-set-favorite__view-favorite"
+						borderless
+						compact
+						onClick={ handleViewFavorites }
+					>
+						{ translate( 'View Favorites' ) }
+					</Button>
+				) }
+			</span>
+		);
+		const options = {
+			id: successNoticeId,
+		};
+		dispatch( removeNotice( successNoticeId ) );
+		dispatch( successNotice( text, options ) );
+	};
+
+	const handleMutation = () => {
+		return {
+			onMutate: async ( { siteId, isFavorite }: { siteId: number; isFavorite: boolean } ) => {
+				// Cancel any current refetches, so they don't overwrite our optimistic update
+				await queryClient.cancelQueries( queryKey );
+
+				// Snapshot the previous value
+				const previousSites = queryClient.getQueryData( queryKey );
+
+				// Optimistically update to the new value
+				queryClient.setQueryData( queryKey, ( oldSites: any ) => {
+					return {
+						...oldSites,
+						sites: oldSites?.sites.map( ( site: Site ) => {
+							if ( site.blog_id === siteId ) {
+								return {
+									...site,
+									is_favorite: ! isFavorite,
+								};
+							}
+							return site;
+						} ),
+					};
+				} );
+
+				// Optimistically update the favorites count of the current query and the sibling query
+				const updateTotalFavorites = ( oldSites: any ) => {
+					return {
+						...oldSites,
+						total_favorites: oldSites.total_favorites + ( isFavorite ? -1 : 1 ),
+					};
+				};
+				queryClient.setQueryData( queryKey, updateTotalFavorites );
+				queryClient.setQueryData( siblingQueryKey, updateTotalFavorites );
+
+				// Store previous settings in case of failure
+				return { previousSites };
+			},
+			onSuccess: () => {
+				handleOnChangeFavoriteSuccess();
+			},
+			onError: ( error: APIError, options: any, context: any ) => {
+				queryClient.setQueryData( queryKey, context?.previousSites );
+				const errorMessage = error.message;
+				dispatch( errorNotice( errorMessage ) );
+			},
+			onSettled: () => {
+				queryClient.invalidateQueries( queryKey );
+			},
+		};
+	};
+
+	const { isLoading, mutate } = useToggleFavoriteSiteMutation( handleMutation() );
+
+	const handleFavoriteChange = () => {
+		mutate( { siteId, isFavorite } );
+		dispatch(
+			recordTracksEvent(
+				isFavorite
+					? 'calypso_jetpack_agency_dashboard_remove_favorite_site'
+					: 'calypso_jetpack_agency_dashboard_set_favorite_site'
+			)
+		);
+	};
+
+	return (
+		<>
```

Comment:
This fragment is unnecessary.

Reply:
Fixed in 44f4bc6fad8841a6816d5b7c41c06a8a85a429ea